02-express-mongo-crud/mongoose-reference-populate/27-28/README.md

```diff
@@ -0,0 +1,359 @@
+ - String
+ - Number
+ - Date
+ - Buffer
+ - Boolean
+ - Mixed
+ - ObjectId  <- used to create referenced data relationships!
+ - Array
+
+[Source Documentation](http://mongoosejs.com/docs/guide.html)  
+
+[SchemaType Documentation](http://mongoosejs.com/docs/schematypes.html)
+
+```js
+// giving mongoose.Schema a shorter name for convenience
+var Schema = mongoose.Schema;
+
+// set up the videogame console schema
+/* Console Schema */
+var consoleSchema = new Schema({
+	name: String,
+	manufacturer: String,
+	released: Date
+});
+```
+
+The `consoleSchema` describes a videogame console such as Nintendo, Sega, or XBox.
+
+
+```js
+/* Game Schema */
+var gameSchema = new Schema({
+	name: String,
+	developer: String,
+	released: Date,
+	// I'm telling consoles to EXPECT references to Console documents
+	consoles: [{type: Schema.Types.ObjectId, ref: 'Console'}]
+});
+```
+The `Game Schema` above describes an actual videogame such as Super Mario Bros., MegaMan, Final Fantasy, and Skyrim.
+
+Note the specific code on line 7 within the `[]` brackets. With the brackets, we're letting the Game Schema know that each game will have an array called `consoles` in it. Inside the `[]`, we're describing what kind of data will go inside a game's `consoles` array as we work with the database. In this case we are telling the Game Schema that we will be filling the `consoles` array with ObjectIds, which is the type of that big beautiful `_id` that Mongo automatically generates for us.
+
+If you forgot, it looks like this: `55e4ce4ae83df339ba2478c6`. That's what's going on with `type: Schema.Types.Objectid`.
+
+When we have the code `ref: 'Console'`, that means that we will be storing ONLY ObjectIds associated with the `Console` document type. Basically, we will only be putting `Console` ObjectIds inside the `consoles` array -- not the whole console object, and not any other kind of data object.
+
+### Make Data with Models
+
+Now that we have our schemas defined, let's compile them all into active models so we can start creating documents!
+
+```js
+/* Compiling models from the above schemas */
+var Game = mongoose.model('Game', gameSchema);
+var Console = mongoose.model('Console', consoleSchema);
+```
+
+Let's make two objects to test out creating a Console document and Game document.
+
+```js
+/* make a new Console document */
+var nin64 = new Console ({
+ name: 'Nintendo 64',
+ manufacturer: 'Nintendo',
+ released: 'September 29, 1996'
+});
+```
+
+```js
+/* make a new Game document */
+var zelda = new Game ({
+  name: 'The Legend of Zelda: Ocarina of Time',
+  developer: 'Nintendo',
+  release: new Date('April 27, 2000'),
+  consoles: []
+});
+```
+
+Notice that we start the `consoles` array empty within our zelda game document. That will be filled with ObjectIds later on.
+
+Now we'll save our work.
+
+```js
+nin64.save(function(err, nintendo64) {
+ if(err) {return console.error(err);}
+ else console.log('nintendo 64 saved successfully');
```

Comment:
You might want to change this syntax and the spacing above.  It's not style-guide compliant.